### ISSUE TITLE
Question about the code

### ISSUE THREAD

**Issue by VasilevIvanVladimirovich** [PBernalPolo/test_MKF#3]
Hello again, I noticed differences in the implementation of java and C++ in filling the matrix M. Can I trust the C++ version?

C++: https://github.com/PBernalPolo/OrientationEstimationTest2017/blob/16a94c82199167ef4b3ead74bd5eceb6df715687/cpp/MEKF.cpp#L164
`  M[0] = F[0];    M[6] = F[3];    M[12] = F[6];    M[18] = F[0]*dt;    M[24] = F[3]*dt;    M[30] = F[6]*dt;
  M[1] = F[1];    M[7] = F[4];    M[13] = F[7];    M[19] = F[1]*dt;    M[25] = F[4]*dt;    M[31] = F[7]*dt;
  M[2] = F[2];    M[8] = F[5];    M[14] = F[8];    M[20] = F[2]*dt;    M[26] = F[5]*dt;    M[32] = F[8]*dt;
  M[3] = 0.0;     M[9] = 0.0;     M[15] = 0.0;     M[21] = 1.0;        M[27] = 0.0;        M[33] = 0.0;
  M[4] = 0.0;     M[10] = 0.0;    M[16] = 0.0;     M[22] = 0.0;        M[28] = 1.0;        M[34] = 0.0;
  M[5] = 0.0;     M[11] = 0.0;    M[17] = 0.0;     M[23] = 0.0;        M[29] = 0.0;        M[35] = 1.0;`
  
java:
https://github.com/PBernalPolo/test_MKF/blob/29c46e17ddc709b18a2365e2c1746f7e78be5c51/test_MKF/MEKF.java#LL78C3-L78C3
 `    M[0] = F[0];    M[6] = F[3];    M[12] = F[6];    M[18] = dt;     M[24] =0.0;     M[30] = 0.0;
    M[1] = F[1];    M[7] = F[4];    M[13] = F[7];    M[19] = 0.0;    M[25] = dt;     M[31] = 0.0;
    M[2] = F[2];    M[8] = F[5];    M[14] = F[8];    M[20] = 0.0;    M[26] = 0.0;    M[32] = dt;
    M[3] = 0.0;     M[9] = 0.0;     M[15] = 0.0;     M[21] = 1.0;    M[27] = 0.0;    M[33] = 0.0;
    M[4] = 0.0;     M[10] = 0.0;    M[16] = 0.0;     M[22] = 0.0;    M[28] = 1.0;    M[34] = 0.0;
    M[5] = 0.0;     M[11] = 0.0;    M[17] = 0.0;     M[23] = 0.0;    M[29] = 0.0;    M[35] = 1.0;`

**Comment by pbernalpolo** [comment]
Hi @VasilevIvanVladimirovich
This one is tricky. The matrix we are looking at is the Jacobian of the evolution equation; this is a matrix that depends on the chosen integration algorithm (there are multiple options). Furthermore, even by choosing a particular integration algorithm, generally the implemented Jacobian matrix is an approximation (a closed expression cannot be obtained in general; one usually gets an infinite sum). That means there is no correct answer; because both are (or should be) approximations. If I had to choose, I'd go with the Java one, because it's the one I wrote at https://www.mdpi.com/1424-8220/19/1/149 eq. (31). However, I wouldn't say that the c++ implementation is totally wrong (I don't remember the logic behind that implementation).
